torch/_dynamo/variables/builder.py

```diff
@@ -461,6 +462,9 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.BUILTIN_MATCH),
             )
+        elif is_utils_checkpoint(value):
+            guards = self.make_guards(GuardBuilder.TYPE_MATCH, GuardBuilder.NAME_MATCH)
```

Comment:
For my understanding, why do we need the guards?

Is it possible to add a test for this?

Reply:
Yes, you are right. I don't think we need this guard. I copied it from the other part where it makes sense because it could map to torch.cond or torch.map or any other higherorder op. But here, its only utils.checkpoint.

I am removing it.